src/raps/execute.ts

```diff
@@ -113,18 +113,31 @@ const delay = (ms: number) => new Promise(res => setTimeout(res, ms));
 
 const waitForNodeAck = async (hash: string, provider: Signer['provider']): Promise<void> => {
   return new Promise(async resolve => {
-    const tx = await provider?.getTransaction(hash);
-    // This means the node is aware of the tx, we're good to go
-    if ((tx && tx.blockNumber === null) || (tx && tx?.blockNumber && tx?.blockNumber > 0)) {
-      resolve();
-    } else {
+    try {
+      const tx = await provider?.getTransaction(hash);
+      // This means the node is aware of the tx, we're good to go
+      if ((tx && tx.blockNumber === null) || (tx && tx?.blockNumber && tx?.blockNumber > 0)) {
+        resolve();
+      } else {
+        // Wait for 1 second and try again
+        await delay(1000);
+        return waitForNodeAck(hash, provider);
+      }
+    } catch (e) {
       // Wait for 1 second and try again
       await delay(1000);
```

Comment:
given that the error case was happening specifically for the swap raps flows which this logic will no longer trigger for, do we need this generic error handling case to infinitely retry? would prefer we only do this retry for specific errors we expect would happen that are valid than just generically